core/block_svg.js

```diff
@@ -689,39 +694,46 @@ Blockly.BlockSvg.prototype.onMouseUp_ = function(e) {
         new Blockly.Events.Ui(this, 'click', undefined, undefined));
   }
+  // This must be before terminateDrag, which deletes the manager.
+  var manager = Blockly.draggedConnectionManager_;
+  if (manager) {
+    var wasDragging = true;
+    var highlightedConnection = manager.closestConnection_;
+    var localConnection = manager.localConnection_;
+    manager.removeConnectionHighlighting();
+  }
   Blockly.terminateDrag_();
 
   var deleteArea = this.workspace.isDeleteArea(e);
 
-  // Connect to a nearby block, but not if it's over the toolbox.
-  if (Blockly.selected && Blockly.highlightedConnection_ &&
-      deleteArea != Blockly.DELETE_AREA_TOOLBOX) {
-    // Connect two blocks together.
-    Blockly.localConnection_.connect(Blockly.highlightedConnection_);
-    if (this.rendered) {
-      // Trigger a connection animation.
-      // Determine which connection is inferior (lower in the source stack).
-      var inferiorConnection = Blockly.localConnection_.isSuperior() ?
-          Blockly.highlightedConnection_ : Blockly.localConnection_;
-      inferiorConnection.getSourceBlock().connectionUiEffect();
-    }
-    if (this.workspace.trashcan) {
-      // Don't throw an object in the trash can if it just got connected.
-      this.workspace.trashcan.close();
-    }
-  } else if (deleteArea && !this.getParent() && Blockly.selected.isDeletable()) {
-    // We didn't connect the block, and it was over the trash can or the
-    // toolbox.  Delete it.
-    var trashcan = this.workspace.trashcan;
-    if (trashcan) {
-      goog.Timer.callOnce(trashcan.close, 100, trashcan);
+  if (wasDragging) {
+    // Connect to a nearby block, but not if it's over the toolbox.
+    if (Blockly.selected && highlightedConnection &&
+        deleteArea != Blockly.DELETE_AREA_TOOLBOX) {
+      // Connect two blocks together.
+      localConnection.connect(highlightedConnection);
+      if (this.rendered) {
+        // Trigger a connection animation.
+        // Determine which connection is inferior (lower in the source stack).
+        var inferiorConnection = localConnection.isSuperior() ?
+            highlightedConnection : localConnection;
+        inferiorConnection.getSourceBlock().connectionUiEffect();
+      }
+      if (this.workspace.trashcan) {
+        // Don't throw an object in the trash can if it just got connected.
+        this.workspace.trashcan.close();
+      }
+    } else if (deleteArea && !this.getParent() && Blockly.selected.isDeletable()) {
+      // We didn't connect the block, and it was over the trash can or the
+      // toolbox.  Delete it.
+      var trashcan = this.workspace.trashcan;
+      if (trashcan) {
+        goog.Timer.callOnce(trashcan.close, 100, trashcan);
+      }
+      Blockly.selected.dispose(false, true);
     }
-    Blockly.selected.dispose(false, true);
-  }
-  if (Blockly.highlightedConnection_) {
-    Blockly.highlightedConnection_.unhighlight();
-    Blockly.highlightedConnection_ = null;
```

Comment:
In trying to understand that this isn't behavior changing but I'm stuck here - can you point me to why Blockly.highlightedConnection_ doesn't have to be set to null anymore? Is it just not used?